pkg/git/libgit2/managed/http.go

```diff
@@ -209,7 +208,32 @@ func createClientRequest(targetUrl string, action git2go.SmartServiceAction, t *
 		return nil, nil, err
 	}
 
+	// Apply authentication and TLS settings to the HTTP transport.
+	if authOpts != nil {
+		if len(authOpts.Username) > 0 {
+			req.SetBasicAuth(authOpts.Username, authOpts.Password)
+			if t.Proxy != nil {
+				t.ProxyConnectHeader.Set(
+					"Authorization",
+					"Basic "+basicAuth(authOpts.Username, authOpts.Password),
```

Comment:
I don't believe it is safe to assume that we always want to pass the credentials to the proxy. Specially for HTTPS, the proxy should not be privy to this information, which is the target website (repository) credentials.

AFAIK, we currently do not support proxy authentication as of yet.

Reply:
We do, the `HTTP_PROXY` variable should have the creds in the URL like `http://user:pswd@proxy.com`